test/e2e/customconfigs/skipjobswithoutqueuename_test.go

```diff
@@ -134,5 +138,122 @@ var _ = ginkgo.Describe("ManageJobsWithoutQueueName", ginkgo.Ordered, func() {
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 		})
+
+		ginkgo.It("should not suspend child jobs of admitted jobs", func() {
+			numPods := 2
+			aw := awtesting.MakeAppWrapper("aw-child", ns.Name).
+				Component(testingjob.MakeJob("job-0", ns.Name).
+					Request(corev1.ResourceCPU, "100m").
+					Parallelism(int32(numPods)).
+					Completions(int32(numPods)).
+					Suspend(false).
+					// Give it enough time that the children would be suspended if parent detection doesn't work as intended.
+					Image(util.E2eTestSleepImage, []string{"3s"}).
+					SetTypeMeta().Obj()).
+				Suspend(false).
+				Obj()
+
+			ginkgo.By("creating an unsuspended appwrapper without a queue name", func() {
+				gomega.Expect(k8sClient.Create(ctx, aw)).To(gomega.Succeed())
+			})
+
+			ginkgo.By("verifying that the appwrapper gets suspended", func() {
+				createdAppWrapper := &awv1beta2.AppWrapper{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(aw), createdAppWrapper)).To(gomega.Succeed())
+					g.Expect(createdAppWrapper.Spec.Suspend).To(gomega.BeTrue())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("setting the queue-name label", func() {
+				createdAppWrapper := &awv1beta2.AppWrapper{}
+				awLookupKey := types.NamespacedName{Name: aw.Name, Namespace: ns.Name}
+				gomega.Eventually(func() error {
+					if err := k8sClient.Get(ctx, awLookupKey, createdAppWrapper); err != nil {
+						return err
+					}
+					createdAppWrapper.Labels["kueue.x-k8s.io/queue-name"] = "main"
+					return k8sClient.Update(ctx, createdAppWrapper)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Wait for appwrapper to be unsuspended", func() {
+				createdAppWrapper := &awv1beta2.AppWrapper{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(aw), createdAppWrapper)).To(gomega.Succeed())
+					g.Expect(createdAppWrapper.Spec.Suspend).To(gomega.BeFalse())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Wait for the wrapped Job to successfully complete", func() {
+				createdAppWrapper := &awv1beta2.AppWrapper{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(aw), createdAppWrapper)).To(gomega.Succeed())
+					g.Expect(createdAppWrapper.Status.Phase).To(gomega.Equal(awv1beta2.AppWrapperSucceeded))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+		})
+
+		ginkgo.It("should not suspend grandchildren jobs of admitted jobs", func() {
```

Comment:
to make sure I'm understanding correctly:

we are testing that after #4251 merges, after the job is admitted, the unsuspend propagates to the job set, and then to the app wrapper?

Reply:
Yes, it is a test case for the bug that #4251 fixes.  The appwrapper gets admitted, which causes it to create the JobSet, which in turn creates the Job, which creates Pods.  We're letting it run for a few seconds (until the Pods complete) to make sure that the [code in the GenericReconciler that handles non-top level jobs ](https://github.com/kubernetes-sigs/kueue/blob/a5e81f751c1a9eb40243457eb80f8af3cfe6576f/pkg/controller/jobframework/reconciler.go#L319-L338)doesn't mistakenly suspend it.  The ancestor job is the AppWrapper, it is admitted, so all of its children should be allowed to run.